Upload kind logs to Github Actions artifact

ashb opened the pull request:
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

potiuk committed:
Send Kubernetes Logs to file.io

ashb committed:
Upload kind logs to Github Artifacts

ashb committed:
fixup! Upload kind logs to Github Artifacts

potiuk commented on .github/workflows/ci.yml:
```diff
@@ -210,6 +210,11 @@ jobs:
       - name: "Deploy app to cluster"
         run: ./scripts/ci/ci_deploy_app_to_kubernetes.sh
+      - uses: actions/upload-artifact@v2
+        if: failure()
```
One small comment. It would be worth while to remove the if here. Those log might be useful to compare and see success vs. failure. And it does not cost us to have them also on success.

ashb replied:
Sure can do.

ashb replied:
(There are some limits on artifacts,  5gb for the whole repo across all builds I think I read somewhere, but if we're only storing compressed text logs it's probably not even close to an issue)

ashb committed:
Always upload Kind logs as GH Actions artifact

ashb committed:
get_ci_environment needed to correctly dump kind logs

Without this the "setup cluster" step dumps the logs as
`kind_logs_2020-06-11_default_default.tar.gz`

Since this is now used outside of just building images, I have moved the
function to _initialization.sh

ashb committed:
Github already compresses artifacts on download.

If we upload a .tar.gz, the download is a .zip file containing that
.tar.gz, which isn't ideal
